### PR TITLE
Documentation for PX-Azure KV secrets integration

### PR DESCRIPTION
**Pull request by ram-infrac**
This PR adds integration of azure key vault(secrets) with the portworx framework

### CODE REVIEW

File: content/key-management/azure-kv/_index.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+title: Azure Key Vault
+logo: /logos/azure-log.png
+keywords: portworx, containers, storage, vault
+description: Instructions on using Azure key vault(Secrets) with Portworx
+weight: 5
```
Comment (adityadani) [inline content/key-management/azure-kv/_index.md]: Change the weight to 7, as 5 is already assigned to Vault

File: content/key-management/azure-kv/_index.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Portworx can integrate with Azure Key Vault [Secrets](https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates#key-vault-secrets) to store your encryption secrets, credentials or passwords. This guide will get a Portworx cluster connected to a Azure Key Vault. The Azure Key Vault could be used to store secrets that will be used for encrypting volumes.
+
+## Setting up Azure Key Vault {#setting-up-azure-kv}
+
+Peruse [this section](https://docs.microsoft.com/en-us/azure/key-vault/quick-create-portal) for help on setting up Azure Key Vault in your setup. 
+You will also require to register and authenticate application with Azure Key Vault. 
+
+- Please follow [doc] (https://docs.microsoft.com/en-us/azure/key-vault/key-vault-manage-with-cli2#registering-an-application-with-azure-active-directory) to register application with azure active directory 
+- Please follow [doc](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-manage-with-cli2#authorizing-an-application-to-use-a-key-or-secret) to grant Azure Key Vault permission to your register app.
```
Comment (adityadani) [inline content/key-management/azure-kv/_index.md]: nit: to your registered app

File: content/key-management/azure-kv/_index.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- Please follow [doc] (https://docs.microsoft.com/en-us/azure/key-vault/key-vault-manage-with-cli2#registering-an-application-with-azure-active-directory) to register application with azure active directory 
+- Please follow [doc](https://docs.microsoft.com/en-us/azure/key-vault/key-vault-manage-with-cli2#authorizing-an-application-to-use-a-key-or-secret) to grant Azure Key Vault permission to your register app. 
+
+Portworx will need application that has Azure Key Vault `set/get/list/delete` secrets permissions. 
+
+Following are the authentication details required by Portworx to connect Azure Key Vault -
+
+- `AZURE_VAULT_URL`: Azure Key Vault URL 
+- `AZURE_TENANT_ID`: Azure Active Directory ID
+- `AZURE_CLIENT_ID`: Azure application ID which is register with Azure active directory and has access to azure key vault mentioned in `AZURE_VAULT_URL`
```
Comment (adityadani) [inline content/key-management/azure-kv/_index.md]: nit: registered

File: content/key-management/azure-kv/_index.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+- `AZURE_CLIENT_SECRET`: Azure application secret id, you may need to generate one if not created already. Follow [doc](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-application-id-and-authentication-key) to generate new secret key for your application
+- `AZURE_ENVIRONMENT`(optional): azure environment i.e [cloudname](https://docs.microsoft.com/en-us/cli/azure/cloud?view=azure-cli-latest#az-cloud-list) AzurePublicCloud(default)
```
Comment (adityadani) [inline content/key-management/azure-kv/_index.md]: nit: Azure environment
Lets not use short forms like i.e and just say OR Cloud Name. Default cloud name is AzurePublicCloud

File: content/key-management/azure-kv/_index.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+1. Use argument `-secret_type azure-kv -cluster_secret_key <secret-id>` when starting Portworx to specify the secret type as vault and the cluster-wide secret key.
+2. Use `-e` docker option to expose the Azure Key Vault enviornment variables.
+
+#### Existing installation
+
+Based on your installation method provide the `-secret_type azure-kv` input argument and environment variable and restart PX on all the nodes.
+
+{{<info>}}
+**Note**: Portworx supports only the [Secrets](https://docs.microsoft.com/en-us/azure/key-vault/about-keys-secrets-and-certificates#key-vault-secrets) of Azure Key Vault
```
Comment (adityadani) [inline content/key-management/azure-kv/_index.md]: I don't understand what this means. May be you can explicitly say what we don't support

File: content/key-management/azure-kv/_index.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+{{</info>}}
+
+### Setting cluster wide secret key
```
Comment (adityadani) [inline content/key-management/azure-kv/_index.md]: Each secret provider has their own set of "uses" series. You can refer to the Google KMS one. Ideally in those docs you can provide info on how to set cluster wide secret key. 

Each uses series has two docs
- One for kubernetes
- One for other schedulers

You can add those as a part of a different PR

Comment (ram-infrac) [reply content/key-management/azure-kv/_index.md]: I missed this or referred old one, will create separate PR for "uses" series

File: content/key-management/azure-kv/_index.md (hunk 5 above)
Comment (ram-infrac) [reply content/key-management/azure-kv/_index.md]: We don't support HSM keys or Softkey by Azure key vault.

Comment (adityadani) [reply content/key-management/azure-kv/_index.md]: Yes I know that, but a reader of this doc, will not understand it. So better to explicitly mention what we don't support

Comment (ram-infrac) [reply content/key-management/azure-kv/_index.md]: I added it as a separate note

**Pull request merged**